### PR TITLE
[serve] Fully remove old handle API

### LINKED ISSUE #43059: [serve] Fully remove old handle API

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Fully remove the previously deprecated `RayServeHandle` and `RayServeSyncHandle` APIs.

## Related issue number

Closes https://github.com/ray-project/ray/issues/43059

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@ray-project/ray-serve need review

File: python/ray/serve/_private/router.py [hunk 1]
```diff
@@ -231,9 +232,7 @@ async def _replace_known_types_in_args(
     ) -> Tuple[Tuple[Any], Dict[str, Any]]:
         """Uses the `_PyObjScanner` to find and replace known types.
 
-        1) Replaces `asyncio.Task` objects with their results. This is used for the old
-           serve handle API and should be removed once that API is deprecated & removed.
-        2) Replaces `DeploymentResponse` objects with their resolved object refs. This
+        1) Replaces `DeploymentResponse` objects with their resolved object refs. This
```
Comment (GeneDer) [inline python/ray/serve/_private/router.py:235]: Maybe no longer need to number it 🙃

**Review by GeneDer — APPROVED** [review]
LGTM!

File: python/ray/serve/_private/router.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/_private/router.py:235]: haha, true!

Comment (edoakes) [reply python/ray/serve/_private/router.py]: fixed

**Review by GeneDer — APPROVED** [review]
:shipit:

**Review by zcin — COMMENTED** [review]
LGTM! Small question about the docs

File: doc/source/serve/api/index.md [hunk 2]
```diff
@@ -42,11 +42,6 @@ This is fixed by added custom filename mappings in `source/conf.py` (look for "a
 
 #### Deployment Handles
 
-:::{note}
-{mod}`DeploymentHandle <ray.serve.handle.DeploymentHandle>` is now the default handle API.
-You can continue using the legacy `RayServeHandle` and `RayServeSyncHandle` APIs using `handle.options(use_new_handle_api=False)` or `export RAY_SERVE_ENABLE_NEW_HANDLE_API=0`, but this support will be removed in a future version.
-:::
-
```
[the remark this answers is not in the document]
Comment (edoakes) [reply doc/source/serve/api/index.md:49]: Great point. I've added a note here and in the model composition section.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**